libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CommonModule } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  NgModule,
+  OnChanges,
+  Output,
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import {
+  ActorDto,
+  EicFunction,
+  MarketRoleDto,
+} from '@energinet-datahub/dh/shared/domain';
+import { TranslocoModule } from '@ngneat/transloco';
+import { MatListModule } from '@angular/material/list';
+import { MarketRoleChanges } from '@energinet-datahub/dh/market-participant/data-access-api';
+import { MarketRoleService } from './market-role-service';
+
+@Component({
+  selector: 'dh-market-participant-actor-market-roles',
+  templateUrl: './dh-market-participant-actor-market-roles.component.html',
+})
+export class DhMarketParticipantActorMarketRolesComponent implements OnChanges {
+  @Input() actor: ActorDto | undefined;
+  @Output() hasChanges = new EventEmitter<MarketRoleChanges>();
+  marketRoleService = new MarketRoleService();
```

Comment:
suggestion: instead of instantiating the class ourselves, let's make Angular do it for us. Remove this line and instead use the `constructor`. Also add the class to the `providers` array that is part of the component's metadata. Example: "dh-market-participant-edit-actor.component.ts#47,62").